vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/dataview/CheckboxGroupListDataViewImpl.java

```diff
@@ -0,0 +1,152 @@
+
+import com.vaadin.flow.component.checkbox.CheckboxGroup;
+import com.vaadin.flow.data.provider.*;
+import com.vaadin.flow.function.SerializableComparator;
+import com.vaadin.flow.function.SerializableConsumer;
+import com.vaadin.flow.function.SerializableFunction;
+import com.vaadin.flow.function.SerializablePredicate;
+import com.vaadin.flow.shared.Registration;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class CheckboxGroupListDataViewImpl<T> implements CheckboxGroupListDataView<T> {
+
+    private int itemsSize = 0;
+    private Set<SizeChangeListener> sizeChangeListeners;
+    private CheckboxGroup<T> component;
+
+    public CheckboxGroupListDataViewImpl(CheckboxGroup<T> checkboxGroup) {
+        this.component = checkboxGroup;
+    }
+
+    @Override
+    public T getNextItem(T item) {
+        return getNextOrPreviousItem(item, itemsList -> {
+            int itemIndex = itemsList.indexOf(item);
+            if (itemIndex != -1 && itemIndex < itemsList.size() - 1) {
+                return itemsList.get(itemIndex + 1);
+            }
+            return null;
+        });
+    }
+
+    @Override
+    public boolean hasNextItem(T item) {
+        return getNextItem(item) != null;
+    }
+
+    @Override
+    public T getPreviousItem(T item) {
+        return getNextOrPreviousItem(item, itemsList -> {
+            int itemIndex = itemsList.indexOf(item);
+            if (itemIndex > 0) {
+                return itemsList.get(itemIndex - 1);
+            }
+            return null;
+        });
+    }
+
+    @Override
+    public boolean hasPreviousItem(T item) {
+        return getPreviousItem(item) != null;
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> withFilter(SerializablePredicate<T> filter) {
+        CheckboxGroupListDataView<T> dataView = withFilterOrOrder(
+                dataProvider -> dataProvider.setFilter(filter));
+        // fire size change event if size has changed
+        getDataSize();
+        return dataView;
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> withSortComparator(SerializableComparator<T> sortComparator) {
+        return withFilterOrOrder(dataProvider -> dataProvider.setSortComparator(sortComparator));
+    }
+
+    @Override
+    public Stream<T> getAllItems() {
+        DataProvider<T, ?> dataProvider = getDataProvider();
+        return dataProvider.fetch(new Query<>());
+    }
+
+    @Override
+    public int getDataSize() {
+        int size = getDataProvider().size(new Query<>());
+        if (size != itemsSize && sizeChangeListeners != null) {
+            sizeChangeListeners.forEach(listener -> listener.sizeChanged(
+                    new SizeChangeEvent<>(this.component, size)));
+            itemsSize = size;
+        }
+        return size;
+    }
+
+    @Override
+    public boolean dataContainsItem(T item) {
```

Comment:
I think this name is against our naming conventions, it should be `isItemPresent(T item)` and I think it should be taking filtering into account (which is doesn't now, but will if `getAllItems()` is changed to return filtered items).

Reply:
Done